### PR TITLE
Update IPAM pool monitor's pool state logic and language

### PR DESCRIPTION
**Pull request by rbtr**
**Reason for Change**:

- Standardize the language for the CNS IP pool:
  - "Allocated": all IPs that CNS has been allocated by DNC.
  - "Assigned": the IPs that CNS has assigned to specific Pods.
  - "Unassigned": allocated - assigned.
- Defines additional data types for modelling IP pool information:
  - `ipPoolState`: is the live view of the IP pool; IPs allocated, assigned, requested, available, pendingProgramming, pendingRelease, unavailable, notInUse.
  - `metaState`: holds pool metadata: the batch size, max size, and min/max free.
  - adds helper methods to build this state from a single query of the CNS pool singleton instead of several locking calls.
- Upcast all the numerics in the pool Monitor to int64 to stop converting and comparing back and forth between int/int64
- Update the IPAM metrics to align with these changes.

**Issue Fixed**:

**Requirements**:

- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests

**Notes**:

### CODE REVIEW

**Review by thatmattlong — APPROVED** [review]

**Comment by rbtr** [comment]
@huntergregory @thatmattlong what happened?

**Comment by huntergregory** [comment]
> @huntergregory @thatmattlong what happened?

Not sure what it means by this:

> huntergregory dismissed thatmattlong’s stale review via 7cc1297 17 hours ago

I made some commits to #1138 last night, but why would that impact this?

**Comment by rbtr** [comment]
Unclear...GitHub thinks you pushed a commit to this branch I think? @thatmattlong no change since your review, do you mind stamping it again

**Review by thatmattlong — APPROVED** [review]

File: cns/ipampool/monitor.go [hunk 1]
```diff
@@ -24,23 +24,24 @@ type nodeNetworkConfigSpecUpdater interface {
 	UpdateSpec(context.Context, *v1alpha.NodeNetworkConfigSpec) (*v1alpha.NodeNetworkConfig, error)
 }
 
-// poolState is the Monitor's view of the IP pool.
-type poolState struct {
-	minFreeCount  int
-	maxFreeCount  int
-	notInUseCount int
+// metaState is the Monitor's configuration state for the IP pool.
+type metaState struct {
```
[the remark this answers is not in the document]
Comment (rbtr) [reply cns/ipampool/monitor.go:28]: will work on this naming as i combine and simplify the pool state structs 👍🏼

**Pull request merged**